ui-core/src/components/inputs/InputWithSuggestions.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {
+  useState,
+  useRef,
+  useEffect,
+  ChangeEventHandler,
+  KeyboardEventHandler,
+  ReactNode,
+  FocusEventHandler,
+} from 'react';
+import cx from 'classnames';
+import Input, { InputProps } from './Input';
+
+interface InputWithSuggestionsProps<T> extends Omit<InputProps, 'onChange' | 'value'> {
+  suggestions: Array<T>;
+  onChange: (value: T) => void;
+  getSuggestionLabel: (option: T) => string;
+  customLabel?: ReactNode;
+  numberOfSuggestionsToShow?: number;
+  exactSearch?: boolean;
+}
+
+const InputWithSuggestions = <T,>({
+  suggestions,
+  onChange,
+  getSuggestionLabel,
+  customLabel,
+  numberOfSuggestionsToShow = 5,
+  exactSearch = false,
+  ...inputProps
+}: InputWithSuggestionsProps<T>) => {
+  const [filteredSuggestions, setFilteredSuggestions] = useState<T[]>([]);
+  const [activeSuggestion, setActiveSuggestion] = useState(-1);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const userInput = e.currentTarget.value.toLowerCase();
+
+    if (userInput === '') {
+      setFilteredSuggestions([]);
+      setShowSuggestions(false);
+      return;
+    }
+
+    const filtered = suggestions.filter((suggestion) => {
+      const suggestionLabel = getSuggestionLabel(suggestion).toLowerCase();
+      return exactSearch
+        ? suggestionLabel.startsWith(userInput)
+        : suggestionLabel.includes(userInput);
+    });
+
+    setFilteredSuggestions(filtered);
+    setShowSuggestions(true);
+  };
+
+  const handleKeyDown: KeyboardEventHandler<HTMLInputElement> = (e) => {
```

Comment:
According to the sketch, when the user navigates through suggestions with the keyboard, the input should display the currently highlighted suggestion.
![image](https://github.com/user-attachments/assets/fa99ac22-3df0-47ba-af38-d9fe69f4fd90)

Reply:
I'm not sure about that. It just seems like the suggestions are sorted alphabetically.

Moreover, this sketch is outdated. You have a new one in the Form section : https://www.sketch.com/s/ca1f66cf-046d-4992-ae63-62187269ba7e/a/3o8Zl5r#Inspect (combo box)

Reply:
Not sure, should i go with alphabetical order ?

Reply:
Still misses the alphabetical order 

![Capture d’écran 2024-08-08 à 11 30 27](https://github.com/user-attachments/assets/3deb3ae3-5229-4897-92ff-0d1a1405e86d)

Reply:
Done :)